### DISCUSSION TITLE
Can we bump the minimum CMake version in the unit test directory?

### DISCUSSION THREAD

**Discussion by falbrechtskirchinger** [nlohmann/json#3376] [Ideas]
Niels, you commented in August of last year, you wanted to avoid CMake 3.13. Is that still a deal breaker even if limited to tests?

I'm working on the C++20 compilation issues and have refactored the CMakeLists.txt such that
```
file(GLOB files src/unit-*.cpp)
foreach(file ${files})
    add_testcases(${file} CXX_STANDARDS default 17 20)
endforeach()
```
generates all required tests/executables (still scanning source files for `JSON_HAS_CPP_*`).

Overrides (like "/STACK:4000000") can then be added via specially named interface libraries.

**Comment by falbrechtskirchinger** [comment]
FYI
```
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0 AND NOT MINGW)
```
is not valid CMake 3.1. I'm fixing it, just wanted to let you know.

**Comment by falbrechtskirchinger** [comment]
Regardless, I do need `target_link_options` from CMake 3.13 to make things work and there doesn't seem to be an alternative. :man_shrugging: 

Waiting for feedback … @nlohmann

**Reply by nlohmann** [reply]
Yes, bumping should be fine.
